network/p2p/p2p_sync.go

```diff
@@ -300,3 +288,42 @@ func (n *p2pNetwork) peersWithProtocolsFilter(protocols ...libp2p_protocol.ID) f
 func allPeersFilter(id peer.ID) bool {
 	return true
 }
+
+func waitSubsetOfPeers(
+	logger *zap.Logger,
+	getSubsetOfPeers func(logger *zap.Logger, vpk spectypes.ValidatorPK, maxPeers int, filter func(peer.ID) bool) (peers []peer.ID, err error),
+	vpk spectypes.ValidatorPK,
+	minPeers, maxPeers int,
+	timeout time.Duration,
+	filter func(peer.ID) bool,
+) ([]peer.ID, error) {
+	if minPeers > maxPeers {
+		return nil, fmt.Errorf("minPeers should not be greater than maxPeers")
+	}
+	if minPeers < 0 || maxPeers < 0 {
+		return nil, fmt.Errorf("minPeers and maxPeers should not be negative")
+	}
+	if timeout <= 0 {
+		return nil, fmt.Errorf("timeout should be positive")
+	}
+
+	// Wait for minPeers with a deadline.
+	deadline := time.Now().Add(timeout)
+	for {
+		peers, err := getSubsetOfPeers(logger, vpk, maxPeers, filter)
+		if err != nil {
+			return nil, err
+		}
+		if len(peers) >= minPeers || minPeers == 0 {
```

Comment:
I don't get why we need to `minPeers == 0` condition if it is never used.

Reply:
leftover from another condition — deleted